nautobot/utilities/testing/__init__.py

```diff
@@ -32,15 +42,55 @@
     "ModelTestCase",
     "ModelViewTestCase",
     "ViewTestCases",
+    "run_job_for_testing",
 )
 
 
+def run_job_for_testing(job, data=None, commit=True, username="test-user", request=None):
+    """Provide a common interface to run Nautobot jobs as part of unit tests."""
```

Comment:
Would be good to flesh out the docstring more, perhaps:

```suggestion
    """
    Provide a common interface to run Nautobot jobs as part of unit tests.
    
    Args:
      job (Job): Job model instance (not Job class) to run
      data (dict): Input data values for any Job variables
      commit (bool): Whether to commit changes to the database or rollback when done.
      username (str): Username of existing or to-be-created User account to own the JobResult
      request (HttpRequest): Existing request (if any) to own the JobResult

    Returns:
      JobResult: representing the executed job
    """
```